src/components/TldrawCanvas.tsx

```diff
@@ -397,51 +394,28 @@ const TldrawCanvas = ({ title }: Props) => {
   const pageUid = useMemo(() => getPageUidByPageTitle(title), [title]);
   const tree = useMemo(() => getBasicTreeByParentUid(pageUid), [pageUid]);
   const appRef = useRef<TldrawApp>();
+  const lastInsertRef = useRef<Vec2dModel>();
   const initialState = useMemo(() => {
     const persisted = getSubTree({
       parentUid: pageUid,
       tree,
       key: "State",
     });
-    const instanceId = TLInstance.createCustomId(pageUid);
-    const userId = TLUser.createCustomId(getCurrentUserUid());
     if (!persisted.uid) {
-      const uid = window.roamAlphaAPI.util.generateUID();
+      // we create a block so that the page is not garbage collected
       createBlock({
         node: {
           text: "State",
-          children: [{ text: "", uid }],
         },
         parentUid: pageUid,
       });
-      return { instanceId, data: undefined, userId, uid };
-    }
-    if (!persisted.children.length) {
-      const uid = window.roamAlphaAPI.util.generateUID();
-      createBlock({
-        node: {
-          text: "",
-          uid,
-        },
-        parentUid: persisted.uid,
-      });
-      return { instanceId, data: undefined, userId, uid };
-    }
-    try {
-      return {
-        data: JSON.parse(window.atob(persisted.children[0].text || "")),
-        instanceId,
-        userId,
-        uid: persisted.children[0].uid,
-      };
-    } catch (e) {
-      return {
-        data: undefined,
-        instanceId,
-        userId,
-        uid: persisted.children[0].uid,
-      };
     }
+    const instanceId = TLInstance.createCustomId(pageUid);
+    const userId = TLUser.createCustomId(getCurrentUserUid());
+    const props = getBlockProps(pageUid);
+    const rjsqb = props["roamjs-query-builder"] as Record<string, unknown>;
+    const data = rjsqb?.tldraw as Parameters<TLStore["deserialize"]>[0];
+    return { instanceId, userId, data };
```

Comment:
Instead of a giant text blob, we are now storing the tldraw data in `block/props`. this makes it queryable!